sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -166,6 +166,15 @@
         });
     }
 
+    ServiceBusReceiverAsyncClient(String fullyQualifiedNamespace, String entityPath, MessagingEntityType entityType,
```

Comment:
Why do we have this?

Reply:
I am just following how Yijun did in acceptSession(), `where a consumer is created  and passed in the constructor` . So  when `ServiceBusReceiverAsyncClient` is created it already have the consumer and it does not create new one.

Reply:
The ServiceBusAsyncClient _contains_ an asyncConsumer.. that's why we have getOrCreate as a child of this.

Reply:
I changed it as we spoke on teams